logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -394,19 +410,18 @@ internal class MLSConversationDataSource(
 
     override suspend fun commitPendingProposals(
         groupID: GroupID
-    ): Either<CoreFailure, Unit> = withContext(serialDispatcher) {
+    ): Either<CoreFailure, Unit> =
```

Comment:
So `commitPendingProposals` is not guarded by the mutex? I suppose because it's also called by other functions in this repository and that would then cause a deadlock?

But `commitpendingProposals` needs a guard. 

How about introducing:
`commitPendingProposals` (mutex)
`internalCommitPendingProposals` (no mutex)